### PR TITLE
[ticket/15392] Changed dirname(__FILE__) to __DIR__

### PR DESCRIPTION
**Pull request by kitsiosk**
Changed `dirname(__FILE__)` to `__DIR__` everywhere

PHPBB3-15392

Checklist:

- [ ] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [ ] Tests pass
- [ ] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [ ] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15392

### CODE REVIEW

File: phpBB/includes/functions.php [hunk 1]
```diff
@@ -3494,12 +3494,12 @@ function phpbb_filter_root_path($errfile)
 	{
 		if ($phpbb_filesystem)
 		{
-			$root_path = $phpbb_filesystem->realpath(dirname(__FILE__) . '/../');
+			$root_path = $phpbb_filesystem->realpath(__DIR__ . '/../');
```
Comment (rubencm) [inline phpBB/includes/functions.php]: you can remove the realpath call

File: build/build_helper.php [hunk 2]
```diff
@@ -43,10 +43,10 @@ function build_package($versions, $verbose = false)
 		$_before = $this->versions[sizeof($this->versions) - 2];
 
 		$this->locations = array(
-			'new_version'	=> dirname(dirname(__FILE__)) . '/phpBB/',
-			'old_versions'	=> dirname(__FILE__) . '/old_versions/',
-			'root'			=> dirname(__FILE__) . '/',
-			'package_dir'	=> dirname(__FILE__) . '/new_version/'
+			'new_version'	=> dirname(__DIR__) . '/phpBB/',
```
Comment (rubencm) [inline build/build_helper.php]: You must remove dirname call here

File: phpBB/includes/functions.php (hunk 1 above)
Comment (DinHere) [reply phpBB/includes/functions.php]: ok fixed

**Comment by DavidIQ** [comment]
Whatever change was done to composer.phar needs to be reverted as it is causing your build to fail. There shouldn't be any change to that file here anyways.

**Comment by DinHere** [comment]
Will it be ok if I copy the composer.phar file from the original repo and try again?

**Comment by DavidIQ** [comment]
Should just be able to do `git checkout origin/3.2.x composer.phar` and commit with proper message format.

File: phpBB/includes/functions.php (hunk 1 above)
Comment (rubencm) [reply phpBB/includes/functions.php]: You have to push your changes again

File: phpBB/phpbb/template/twig/loader.php [hunk 3]
```diff
@@ -35,7 +35,7 @@ public function __construct(\phpbb\filesystem\filesystem_interface $filesystem,
 	{
 		$this->filesystem = $filesystem;
 
-		parent::__construct($paths, $this->filesystem->realpath(dirname(__FILE__)));
+		parent::__construct($paths, $this->filesystem-> __DIR__);
```
Comment (rubencm) [inline phpBB/phpbb/template/twig/loader.php]: parent::__construct(__DIR__);

**Comment by rubencm** [comment]
You have to remove only realpaths that only have __DIR__ as argument, because __DIR__ == realpath(__DIR__), but not all of them

**Comment by DinHere** [comment]
@rubencm I only see one line where realpath has only __DIR__ as arguement, in this file: phpBB/phpbb/template/twig/loader.php. Do I only have to change this file?

**Comment by rubencm** [comment]
Give me a minute, i will comment the lines

File: phpBB/develop/namespacify.php [hunk 4]
```diff
@@ -20,10 +20,10 @@
 //
 die("Please read the first lines of this script for instructions on how to enable it");
 
-$namespace_dir = realpath(__DIR__ . '/../phpbb/');
-$code_dir = realpath(__DIR__ . '/../');
-$test_dir = realpath(__DIR__ . '/../../tests/');
-$config_dir = realpath(__DIR__ . '/../config/');
+$namespace_dir = __DIR__ . '/../phpbb/';
```
Comment (rubencm) [inline phpBB/develop/namespacify.php]: Dont remove realpath in this lines

File: phpBB/develop/rename_interfaces.php [hunk 5]
```diff
@@ -20,8 +20,8 @@
 //
 die("Please read the first lines of this script for instructions on how to enable it");
 
-$code_dir = realpath(__DIR__ . '/../');
-$test_dir = realpath(__DIR__ . '/../../tests/');
+$code_dir = __DIR__ . '/../';
```
Comment (rubencm) [inline phpBB/develop/rename_interfaces.php]: Dont remove realpath in this lines

File: phpBB/includes/functions.php [hunk 6]
```diff
@@ -3494,12 +3494,12 @@ function phpbb_filter_root_path($errfile)
 	{
 		if ($phpbb_filesystem)
 		{
-			$root_path = $phpbb_filesystem->realpath(dirname(__FILE__) . '/../');
+			$root_path = $phpbb_filesystem-> __DIR__ . '/../';
```
Comment (rubencm) [inline phpBB/includes/functions.php]: Dont remove realpath here

File: phpBB/includes/functions.php [hunk 7]
```diff
@@ -3494,12 +3494,12 @@ function phpbb_filter_root_path($errfile)
...
 		}
 		else
 		{
 			$filesystem = new \phpbb\filesystem\filesystem();
-			$root_path = $filesystem->realpath(dirname(__FILE__) . '/../');
+			$root_path = $filesystem-> __DIR__ . '/../';
```
Comment (rubencm) [inline phpBB/includes/functions.php]: Dont remove realpath here

**Comment by DinHere** [comment]
Ok I now understand my mistake to remove realpath everywhere :/. I hope it now passes the test

File: Vagrantfile [hunk 8]
```diff
@@ -2,7 +2,7 @@ require 'json'
 require 'yaml'
 
 VAGRANTFILE_API_VERSION ||= "2"
-confDir = $confDir ||= File.expand_path("phpBB/vendor/laravel/homestead", File.dirname(__FILE__))
+confDir = $confDir ||= File.expand_path("phpBB/vendor/laravel/homestead", File.__DIR__)
```
Comment (rubencm) [inline Vagrantfile]: Leave this as it was before, this is not php

File: phpBB/includes/functions.php [hunk 9]
```diff
@@ -3494,12 +3494,12 @@ function phpbb_filter_root_path($errfile)
 	{
 		if ($phpbb_filesystem)
 		{
-			$root_path = $phpbb_filesystem->realpath(dirname(__FILE__) . '/../');
+			$root_path = $phpbb_filesystem-> realpath(__DIR__ . '/../');
```
Comment (rubencm) [inline phpBB/includes/functions.php]: Remove space between $phpbb_filesystem-> and realpath(__DIR__ . '/../');

File: phpBB/includes/functions.php [hunk 10]
```diff
@@ -3494,12 +3494,12 @@ function phpbb_filter_root_path($errfile)
...
 		}
 		else
 		{
 			$filesystem = new \phpbb\filesystem\filesystem();
-			$root_path = $filesystem->realpath(dirname(__FILE__) . '/../');
+			$root_path = $filesystem-> realpath(__DIR__ . '/../');
```
Comment (rubencm) [inline phpBB/includes/functions.php]: Remove space between $phpbb_filesystem-> and realpath(__DIR__ . '/../');

**Comment by rubencm** [comment]
Now it passes the test again :+1:

File: Vagrantfile [hunk 11]
```diff
@@ -2,7 +2,7 @@ require 'json'
 require 'yaml'
 
 VAGRANTFILE_API_VERSION ||= "2"
-confDir = $confDir ||= File.expand_path("phpBB/vendor/laravel/homestead", File.dirname(__FILE__))
+confDir = $confDir ||= File.expand_path("phpBB/vendor/laravel/homestead", File.dirname(__DIR__))
```
Comment (rubencm) [inline Vagrantfile]: File.dirname(__FILE__)

File: phpBB/includes/functions.php [hunk 12]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (rubencm) [inline phpBB/includes/functions.php]: Remove this

**Comment by mrgoldy** [comment]
Hello, I am not sure whether you are still active. But this PR will need some work before it can be merged, as it slipped through for quite some time I see. The conflicts will need to be resolved (by running a rebase) and the commit messages have to be amended or squashed, as they will not pass tests. If you need any assistance, please let me know and I'll see if I can guide you through it. If you say "No thank you I'm done", that's completely understandable aswell, then if you'll allow me I'll do it for you.

**Comment by marc1706** [comment]
Replaced by https://github.com/phpbb/phpbb/pull/6150

**Pull request closed without merging**